com.unity.netcode.gameobjects/Runtime/Transports/UTP/UnityTransport.cs

```diff
@@ -707,6 +667,38 @@ private bool ProcessEvent()
         private void Update()
         {
             if (IsDisabledBySimulator)
+            {
+                // Flush all send queue items
+                foreach (var queueItem in m_SendQueue)
+                {
+                    queueItem.Value.Consume(queueItem.Value.Length);
+                }
```

Comment:
Note that for reliable traffic, this is will not correctly simulate the behavior of a network disconnection. Normally during a disconnection, reliable acknowledgements will not make it back, and so reliable traffic will keep piling up in the send queues until they overflow (which raises a disconnect event). A more accurate simulation would be to flush unreliable send queues, and just avoid calling `SendBatchedMessages` for reliable queues.

Reply:
I was going to ask you about this, but you're faster than my fingers! :D Thanks for the info I'll fix as suggested

Reply:
Should we just early return out of the update loop without proactively clearing any queues or events? It seems like a more robust implementation would involve adding an entrypoint to do something in UTP

Reply:
Yeah, I think at this point we're probably better off finding the "good enough" approach, while we discuss next steps for something more robust. I'll revert the changes and schedule a chat, though it might be a little bit since I'm off next week and Beck the next.